### ISSUE TITLE
on JDK 9+, build definition consistently fails to load on second try

### ISSUE THREAD

**Issue by SethTisue** [scalameter/scalameter#204]
I've figured out how to reproduce the error I also mentioned in #203:

```
[scalameter] [error] /home/jenkins/workspace/scala-2.12.x-jdk9-integrate-community-build/target-0.9.12/sbt-base-dir/extractor/staging/44ec20cefe008edf2897/mecha/src/main/scala/org/stormenroute/mecha/source-control.scala:156: type mismatch;
[scalameter] [error]  found   : org.stormenroute.mecha.org.stormenroute.mecha.BufferedLogger
[scalameter] [error]  required: org.stormenroute.mecha.(some other)org.stormenroute.mecha.BufferedLogger
[scalameter] [error]       (name, logger)
[scalameter] [error]              ^
[scalameter] [error] /home/jenkins/workspace/scala-2.12.x-jdk9-integrate-community-build/target-0.9.12/sbt-base-dir/extractor/staging/44ec20cefe008edf2897/mecha/src/main/scala/org/stormenroute/mecha/source-control.scala:170: type mismatch;
[scalameter] [error]  found   : org.stormenroute.mecha.org.stormenroute.mecha.BufferedLogger
[scalameter] [error]  required: org.stormenroute.mecha.(some other)org.stormenroute.mecha.BufferedLogger
[scalameter] [error]       (name, logger)
[scalameter] [error]              ^
```

it doesn't happen on JDK 8, only on 9 or 10.

it doesn't happen until the second time you run sbt. after that, it happens every time.

you can make the problem go away temporarily by running `rm -rf ~/.sbt/0.13/staging`. this seems to return things (specifically, the source dependency on `mecha`) to a clean enough state to make it work again for one run.

**Comment by SethTisue** [comment]
https://github.com/scala/community-builds/issues/747 is about this issue in the context of the Scala community build.

**Comment by SethTisue** [comment]
@eed3si9n @dwijnand does the compilation error look like anything you've seen happen before with source dependencies in sbt?

**Comment by eed3si9n** [comment]
I've never seen it before.

**Comment by dwijnand** [comment]
I've seen issues to do with different parts of sbt messing with the thread context classloader leading to confusions. But not this, or any issue reproducing as reliably as the one you're describing here.

**Comment by SethTisue** [comment]
resolved by using a binary dependency instead (9924aad2e7cc47927e6270320d02307be07fdee3)
